### DISCUSSION TITLE
Enable FT800-FT813 library for STM32F4

### DISCUSSION THREAD

**Discussion by joetrontria** [RudolphRiedel/FT800-FT813#25] [Q&A]
Hi Rudolph,
Thank you so much for your sharing the library.  Can you please give me some pointer to see where is the problem reside?
I am using STM32F4 39ZI, the library migration is straight forward.
I created the project from cube so it is using normal  MX_SPI1_Init(); to init the SPI hence I don't need to call EVE_init_spi(void); mater of fact the library didn't have the EVE_init_spi() function for the STM32F4 option.
My project doesn't have the stm32f4xx_ll_spi.h/c but I manage to download the driver.
Code is compiled successfully no problem, but when I started to run it stuck at this while(!LL_SPI_IsActiveFlag_TXE(EVE_SPI)); in the function spi_transmit(uint8_t data).
I connected the signals to logic analyzer but I don't see clk, miso, mosi data.
I double check the library is using SPI1 as my MX_SPI1_Init().
Am I missing something or did something wrong in the configuration?
Thank you so much for your help.

**Comment by RudolphRiedel** [comment]
Hello,

I am really in no position to give advice on any STM32 related issues since I never actively used any STM32.

That said, I dug out my STM32 test project I worked on a while ago, updated it to the latest files and updated EVE_target.c and EVE_target.h with the small improvements over the released code I found.

Attached is my test project for PlatformIO.
This builds for a whole bunch of different STM32 targets but so far this is merely an exercise in making it compile for these
different targets as it does not do anything usefull yet.

And this also reminded me that I have not re-installed STM32CubeIDE yet, I recently updated my system.

[EVE_Test_STM32_RiTFT50.zip](https://github.com/RudolphRiedel/FT800-FT813/files/6923014/EVE_Test_STM32_RiTFT50.zip)

**Comment by joetrontria** [comment]
Hi Rudolph,
Thank you for your reply.  Your STM32 project is pretty much the same of my project setting.  Your code will compile no problem like you said.
I just want to clarify if you intended to compile without the DMA is enable because in the source code I saw you still have the 
//		#define EVE_DMA		/* do not activate, it is not working yet */
commented out.
so your code will not have any SPI initialization because you didn't call EVE_init_spi() nor enable the DMA initialization.

I used your same library and it is working perfectly fine with ESP32 for PlatformIO and in ESP32 I had to call EVE_init_spi() to associate the SPI port with your library.

There is still something that missing for me that my SPI bus didn't get call to send out the data.

**Reply by RudolphRiedel** [reply]
I intended to fix this but never got to actually fixing it.
And right now I could not even fix it since I left the only STM32 eval board I have at work.

**Reply by joetrontria** [reply]
Hi Rudolph,
Can you please give me your address? I will send you couple of the STM32 board as the appreciation for all the handworks you put in this library.  They are free from me since I have extras STM32 eval boards from different families

**Reply by RudolphRiedel** [reply]
Thank you for your offer but getting STM32 eval boards is no issue at all.
I can not use STM32 controllers for anything though which is why I never got deeper involved with these.
The main reason is that there are no automotive versions of STM32 controllers, this renders them mostly irrelevant in my line of work.
There are two projects in which I could use a STM32 controller and would like to design boards for but none of the few parts that would be useable are available so these are put on hold. And while the one eval board I have does have the right controller for one of the projects, the one feature needing the most testing is not available - USB.

So it is not that I am not interested in supporting STM32, I just feel that I can't right now.
I will retrieve my 446RE board from work though.

**Reply by joetrontria** [reply]
No problem at all.  Please just let me know if you can get your library working on STM32 on PlatformIO just by display a text is good enough for my understanding.

**Comment by RudolphRiedel** [comment]
Well now, technically it is working, at least mostly except for supporting DMA.
The spi init and everything else to bring up the controller is not per se part of the library.
Things like the clock configuration are very target specific.
And with the ARDUINO_NUCLEO_F446RE target I am using the same low-level HAL driver from STM.
Is this the end of it? No, I still would like to make my native example application work.
But this is just the way it currently is and if you like to you could contribute to a solution for this.

**Comment by MahmutKocamann** [comment]
[RudolphRiedel](https://github.com/RudolphRiedel)  Thank you very much, Mr. Rudolph Riedel. I'm using FT811 driver display and I haven't been able to spi communication for a long time. I've been trying to figure out where I went wrong for days and I solved my problem with the code you shared. For now at least I can read the ID. Thank you very much for sharing this valuable information. I wish you good work.

**Reply by RudolphRiedel** [reply]
Thank you!
I had my fair share of issues, especially in the beginning. :-)
And well, STM32 and the chip-crisis, I would like to design a PCB with a STM32H7B0RBT6 but, well.
It is even possible that they "launched" a different chip by now that would be a match for the project.
So it is not so much that I do not like STM32, it is more that they only offer a very limited range of controllers that I could actually use for my projects and these are unicorns.
